core/runtime.rs

```diff
@@ -716,6 +818,8 @@ impl JsRuntime {
       .expect("ModuleInfo not found");
     let mut status = module.get_status();
 
+    let (sender, receiver) = mpsc::channel(1);
```

Comment:
* Using a channel within a thread ... not my favorite pattern ever.
* Wouldn't a Oneshot work just as well?

Reply:
> * Using a channel within a thread ... not my favorite pattern ever.

That's the only solution I came up with.

> * Wouldn't a Oneshot work just as well?

I tried, but couldn't get it past borrow checker on `sender` not being clonable. I'll try to address this in the next pass.

Reply:
Yeah I don't have any suggestions either...